ludwig/utils/data_utils.py

```diff
@@ -1089,3 +1089,17 @@ def use_credentials(creds):
                     json.dump(old_conf, f)
                 conf.clear()
                 set_conf_files(tmpdir, conf)
+
+
+def get_sanitized_feature_name(feature_name: str) -> str:
+    """Replaces non-word characters (anything other than alphanumeric or _) with _.
+
+    Used in model config initialization and sanitize_column_names(), which is called during dataset building.
+    """
+    return re.sub(r"[(){}.:]", "_", feature_name)
```

Comment:
I think we need to add `[` and `]` as well, both of those would likely break LightGBM. Do we want to sub whitespace too?

Reply:
We should probably include quotation marks as well.

Reply:
Done. I didn't include whitespace because this would break a backwards compatibility test, and I also haven't seen whitespace cause issues.

Reply:
That makes sense. Thanks for adding those in!